src/plugins/intel_gpu/src/graph/reduce.cpp

```diff
@@ -95,6 +97,117 @@ layout reduce_inst::calc_output_layout(reduce_node const& node, kernel_impl_para
+    auto axes_tensor = std::make_shared<ngraph::runtime::HostTensor>(ov::element::i64, ov::Shape{axes.size()}, axes.data());
+    std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> const_data = {{1, axes_tensor}};
+
+    // shape infer by mode
+    auto mode = desc->mode;
+    switch (mode) {
+        case reduce_mode::max:
+        {
+            ov::op::v1::ReduceMax op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::min:
+        {
+            ov::op::v1::ReduceMin op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::mean:
+        {
+            ov::op::v1::ReduceMean op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::prod:
+        {
+            ov::op::v1::ReduceProd op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum:
+        {
+            ov::op::v1::ReduceSum op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_and:
+            {
+            ov::op::v1::ReduceLogicalAnd op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_or:
+        {
+            ov::op::v1::ReduceLogicalOr op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l1:
+        {
+            ov::op::v4::ReduceL1 op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l2:
+        {
+            ov::op::v4::ReduceL2 op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum_square:
+            // not implemented
+        case reduce_mode::log_sum:
+            // not implemented
+        case reduce_mode::log_sum_exp:
+            // not implemented
+        default:
+            CLDNN_ERROR_MESSAGE(
+                desc->id,
+                "Not supported mode");
+    }
+
+    // get 'output_type'
+    auto output_type = input0_layout.data_type;
+    std::vector<reduce_mode> reduce_bool_modes = {reduce_mode::logical_and, reduce_mode::logical_or};
+    if (std::find(reduce_bool_modes.begin(), reduce_bool_modes.end(), mode) != reduce_bool_modes.end())
+        output_type = data_types::i8;
+    else if (output_type == data_types::i8 || output_type == data_types::u8)
+        output_type = data_types::f32;
+
+    if (desc->output_data_type)
+        output_type = *desc->output_data_type;
+
+    if (impl_param.has_fused_primitives())
+        output_type = impl_param.get_fused_output_layout().data_type;
```

Comment:
If i8 => fp32 was due to the overflow, this should be changed if the fused_output type is int8?